modules/seqexec/engine/src/main/scala/seqexec/engine/Engine.scala

```diff
@@ -407,8 +415,8 @@ object Engine {
 
   object State {
     def empty: State = State(Map.empty)
-    def atSequence(id: Observation.Id): Lens[Map[Observation.Id, Sequence.State], Option[Sequence.State]] = at(id)
-    def sequenceState[D](id: Observation.Id): Lens[State, Option[Sequence.State]] = State.sequences ^|-> atSequence(id)
+    def atSequence(id: Observation.Id): Optional[Map[Observation.Id, Sequence.State], Sequence.State] = index(id)
```

Comment:
Just taking advantage of monocle.Optional

Reply:
Did you add tests to these?